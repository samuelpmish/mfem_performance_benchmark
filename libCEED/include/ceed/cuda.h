// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and other CEED contributors.
// All Rights Reserved. See the top-level LICENSE and NOTICE files for details.
//
// SPDX-License-Identifier: BSD-2-Clause
//
// This file is part of CEED:  http://github.com/ceed

#ifndef CEED_CUDA_H
#define CEED_CUDA_H

#include <ceed.h>
#include <cuda.h>

CEED_EXTERN int CeedQFunctionSetCUDAUserFunction(CeedQFunction qf, CUfunction f);

#endif  // CEED_CUDA_H
