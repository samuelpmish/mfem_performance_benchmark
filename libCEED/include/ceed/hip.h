// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and other CEED contributors.
// All Rights Reserved. See the top-level LICENSE and NOTICE files for details.
//
// SPDX-License-Identifier: BSD-2-Clause
//
// This file is part of CEED:  http://github.com/ceed

#ifndef CEED_HIP_H
#define CEED_HIP_H

#include <ceed.h>
#include <hip/hip_runtime.h>

CEED_EXTERN int CeedQFunctionSetHIPUserFunction(CeedQFunction qf, hipFunction_t f);

#endif  // CEED_HIP_H
