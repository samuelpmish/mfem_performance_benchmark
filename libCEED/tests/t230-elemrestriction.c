/// @file
/// Test creation and view of an element restriction at points
/// \test Test creation and view of an element restriction at points
#include <ceed.h>

int main(int argc, char **argv) {
  Ceed                ceed;
  CeedInt             num_elem = 3, num_points = num_elem * 2;
  CeedInt             ind[(num_elem + 1) + num_points];
  CeedElemRestriction elem_restriction;

  CeedInit(argv[1], &ceed);

  {
    CeedInt offset      = num_elem + 1;
    CeedInt point_index = num_elem;

    for (CeedInt i = 0; i < num_elem; i++) {
      CeedInt num_points_in_elem = (i + 1) % num_elem + 1;

      ind[i] = offset;
      for (CeedInt j = 0; j < num_points_in_elem; j++) {
        ind[offset + j] = point_index;
        point_index     = (point_index + 1) % num_points;
      }
      offset += num_points_in_elem;
    }
    ind[num_elem] = offset;
  }
  CeedElemRestrictionCreateAtPoints(ceed, num_elem, num_points, 1, num_points, CEED_MEM_HOST, CEED_USE_POINTER, ind, &elem_restriction);

  CeedElemRestrictionView(elem_restriction, stdout);

  CeedElemRestrictionDestroy(&elem_restriction);
  CeedDestroy(&ceed);
  return 0;
}
