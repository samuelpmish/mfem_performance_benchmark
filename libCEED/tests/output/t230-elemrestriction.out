CeedElemRestriction at points from (6, 1) to 3 elements with a maximum of 3 points on an element
